src/System.Threading.Tasks.Extensions/src/System/Runtime/CompilerServices/AsyncBuilderAttribute.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Indicates the async method builder that should be used by the C# compiler to build
+    /// the attributed type when used as the return type of an async method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct, Inherited = false, AllowMultiple = false)]
```

Comment:
Technically it only needs Struct, since that's all it's used for in this assembly.  I can add Interface for completeness, though, and to make this more easily copy-and-pasteable in case others want to use it.